src/blocks/music.rs

```diff
@@ -287,13 +295,19 @@ impl ConfigBlock for Music {
                 500,
             )
             .unwrap();
+
+        let mut preferred_players = Vec::new();
+
+        if let Some(player) = block_config.player {
+            match player {
+                PlayerName::PlayerName(player) => preferred_players.push(player.clone()),
+                PlayerName::PlayerNames(players) => preferred_players.extend(players.clone()),
```

Comment:
You can avoid extra allocations here: `preferred_players = players` should be sufficient (I haven't tested this).